src/renderer/gdi/state.cpp

```diff
@@ -344,85 +344,105 @@ GdiEngine::~GdiEngine()
-
-        // if the limit is <= 0 (no height at all), stick with the desired height.
-        // This is how we force a curlyline even when there's no space, though it
-        // might be clipped at the bottom.
-        if (maxDrawableCurlyLinePeakHeight > 0.0f)
-        {
-            _lineMetrics.curlylinePeakHeight = std::min(_lineMetrics.curlylinePeakHeight, maxDrawableCurlyLinePeakHeight);
-        }
-    }
+    const auto underlineOffset = std::roundf(idealUnderlineOffset);
+    const auto underlineWidth = std::max(1.0f, std::roundf(idealUnderlineWidth));
+    const auto strikethroughOffset = std::roundf(idealStrikethroughOffset);
+    const auto strikethroughWidth = std::max(1.0f, std::roundf(idealStrikethroughWidth));
+    const auto thinLineWidth = std::max(1.0f, std::roundf(idealUnderlineWidth / 2.0f));
+
+    // For double underlines we loosely follow what Word does:
+    // 1. The lines are half the width of an underline (= thinLineWidth)
+    // 2. Ideally the bottom line is aligned with the bottom of the underline
+    // 3. The top underline is vertically in the middle between baseline and ideal bottom underline
+    // 4. If the top line gets too close to the baseline the underlines are shifted downwards
+    // 5. The minimum gap between the two lines appears to be similar to Tex (1.2pt)
+    // (Additional notes below.)
+
+    // 2.
+    auto doubleUnderlinePosBottom = underlineOffset + underlineWidth - thinLineWidth;
+    // 3. Since we don't align the center of our two lines, but rather the top borders
+    //    we need to subtract half a line width from our center point.
+    auto doubleUnderlinePosTop = std::roundf((baseline + doubleUnderlinePosBottom - thinLineWidth) / 2.0f);
+    // 4.
+    doubleUnderlinePosTop = std::max(doubleUnderlinePosTop, baseline + thinLineWidth);
+    // 5. The gap is only the distance _between_ the lines, but we need the distance from the
+    //    top border of the top and bottom lines, which includes an additional line width.
+    const auto doubleUnderlineGap = std::max(1.0f, std::roundf(1.2f / 72.0f * _iCurrentDpi));
+    doubleUnderlinePosBottom = std::max(doubleUnderlinePosBottom, doubleUnderlinePosTop + doubleUnderlineGap + thinLineWidth);
+    // Our cells can't overlap each other so we additionally clamp the bottom line to be inside the cell boundaries.
+    doubleUnderlinePosBottom = std::min(doubleUnderlinePosBottom, cellHeight - thinLineWidth);
+
+    // The wave line is drawn using a cubic Bézier curve (PolyBezier), because that happens to be cheap with GDI.
+    // We use a Bézier curve where, if the start (a) and end (c) points are at (0,0) and (1,0), the control points are
+    // at (0.5,0.5) (b) and (0.5,-0.5) (d) respectively. Like this but a/b/c/d are square and the lines are round:
+    //
+    //       b
+    //
+    //     ^
+    //    / \                here's some text so the compiler ignores the trailing \ character
+    //   a   \   c

```

Comment:
i assume this should be 
```suggestion
    //   a   \   d
```

because line R413 has `c` on it